src/browser/index.ts

```diff
@@ -11,20 +11,41 @@
  */
 
 import { AxiosInstance } from 'axios';
+import { devworkspaceIdentifier } from '../common';
 import {
   IDevWorkspaceApi,
   IDevWorkspaceClientApi,
+  IDevWorkspaceTemplateApi,
 } from '../types';
+import { findApi } from './helper';
 import { RestDevWorkspaceApi } from './workspace-api';
+import { RestDevWorkspaceTemplateApi } from './template-api';
 
 export class RestApi implements IDevWorkspaceClientApi {
+  private axios: AxiosInstance;
   private _workspaceApi: IDevWorkspaceApi;
+  private _templateApi: IDevWorkspaceTemplateApi;
+  private apiEnabled: boolean | undefined;
 
   constructor(axios: AxiosInstance) {
+    this.axios = axios;
     this._workspaceApi = new RestDevWorkspaceApi(axios);
+    this._templateApi = new RestDevWorkspaceTemplateApi(axios);
   }
 
   get workspaceApi(): IDevWorkspaceApi {
     return this._workspaceApi;
   }
+
+  get templateApi(): IDevWorkspaceTemplateApi {
+    return this._templateApi;
+  }
+
+  async isDevWorkspaceResourceEnabled(): Promise<boolean> {
+    if (this.apiEnabled !== undefined) {
+      return Promise.resolve(this.apiEnabled);
+    }
+    this.apiEnabled = await findApi(this.axios, devworkspaceIdentifier);
+    return this.apiEnabled;
```

Comment:
AFAIU Probably here we need to check v1alpha2 if possible, otherwise this check will be true when WTO is installed but integration with Che is not available since it relies on v1alpha2.